src/OpenSage.Game/Logic/Object/GameObject.cs

```diff
@@ -545,12 +561,7 @@ public bool UpgradeAvailable(UpgradeTemplate upgrade)
 
         public bool ConflictingUpgradeAvailable(UpgradeTemplate upgrade)
         {
-            if (upgrade == null)
-            {
-                return false;
-            }
-
-            if (upgrade.Type == UpgradeType.Player)
+            if (upgrade == null || upgrade.Type == UpgradeType.Player)
```

Comment:
When UpgradeType is Player, it should probably check the owner for conflicting upgrades

Reply:
There is already a todo for that in the line below, just combined both checks into a single line here